### DISCUSSION TITLE
Bulk Deletion Issue in Filament: Records Deleted Incorrectly

### DISCUSSION THREAD

**Discussion by ghassanhjr** [filamentphp/filament#8332] [General]
in the page of  filament documentation [ Deleting records ](https://filamentphp.com/docs/2.x/admin/resources/deleting-records)

_They also have the ability to bulk-delete records if the deleteAny() method of the policy returns true. Filament uses the deleteAny() method because iterating through multiple records and checking the delete() policy is not very performant._

**The problem I am facing:**
When I do Bulk action in the list table for either one record or multi that are not created by the user in step, it directly deletes them; here is some focused code for you to refer to.

And this is my policy code. 
```
public function delete(User $user, Commodity $commodity): bool
    {
        // Check if the authenticated user's ID matches the user ID associated with the commodity.
        // This indicates that the user is the creator of the commodity and can view it.
        return $user->id === $commodity->user_id;
    }
```

And this is the resource bulk action,
```
->bulkActions(
                [
                    Tables\Actions\BulkActionGroup::make(
                        [
                           
                            Tables\Actions\DeleteBulkAction::make(),
                        ]
                    ),

                ]
```
![image](https://github.com/filamentphp/filament/assets/90517109/6987a906-995d-4403-86ca-9a3f81e6d43f)

**Comment by danharrin** [comment]
Yeah, if you only allow users to delete some records then you should return false from `deleteAny()` (hence **ANY**, there are no restrictions), or override `DeleteBulkAction::make()->action()` to perform checks on each record before deleting
